debian/pi-bluetooth.bthelper@.service

```diff
@@ -2,7 +2,7 @@
 Description=Raspberry Pi bluetooth helper
 Requires=hciuart.service
 After=hciuart.service
-Before=bluetooth.service
+After=bluetooth.service
```

Comment:
I'll need a lot of persuading to accept this change. Running bthelper, which potentially changes the BDADDR, before starting the bluetooth service can avoid a variety of problems.

Reply:
Please describe your configuration - model of Pi, the Ethernet MAC address, whether you are using an external Bluetooth dongle, etc.

Reply:
It is a Pi 4 with 4GB RAM, Kernel 5.10.17-v7l+. I'm using the internal Bluetooth module, no dongles except a USB ZigBee dongle (ConBee 2) in case that's relevant. 
The Ethernet mac address is dc:a6:32:1c:6c:67
I'm not using Ethernet though, just the internal WiFi.

I see your point regarding bthelper. Frankly, I've just found that bluetooth.service did not start reliably without that change.

Reply:
To help diagnose the problem, comment out the line in /usr/bin/bthelper that calls bluetoothctl to toggle the power off and on. Unless you are using Secure Simple Pairing it shouldn't make a difference to functionality, but it might help it start up.

Reply:
@pelwell I commented out line 41 in `/usr/bin/bthelper`:
`(sleep 5; /usr/bin/bluetoothctl power off; /usr/bin/bluetoothctl power on) &`
reverted the change to `/lib/systemd/system/bthelper@.service` and rebooted. `bluetooth.service` did not start again unfortunately. I see the following with `journalctl -xe` but nothing about `bluetooth.service`

```
May 20 08:24:49 raspberrypi systemd[1]: Starting Raspberry Pi bluetooth helper...
-- Subject: A start job for unit bthelper@hci0.service has begun execution
-- Defined-By: systemd
-- Support: https://www.debian.org/support
-- 
-- A start job for unit bthelper@hci0.service has begun execution.
-- 
-- The job identifier is 285.
May 20 08:24:49 raspberrypi bthelper[529]: Raspberry Pi BDADDR already set

```

Reply:
In case it is relevant, here is my cpuinfo. In #23 @jaharkema reported an issue and I noticed they had a different hardware rev of Pi 4 Model B (1.2 compared to my 1.1)

```
processor       : 0
model name      : ARMv7 Processor rev 3 (v7l)
BogoMIPS        : 108.00
Features        : half thumb fastmult vfp edsp neon vfpv3 tls vfpv4 idiva idivt vfpd32 lpae evtstrm crc32 
CPU implementer : 0x41
CPU architecture: 7
CPU variant     : 0x0
CPU part        : 0xd08
CPU revision    : 3

processor       : 1
model name      : ARMv7 Processor rev 3 (v7l)
BogoMIPS        : 108.00
Features        : half thumb fastmult vfp edsp neon vfpv3 tls vfpv4 idiva idivt vfpd32 lpae evtstrm crc32 
CPU implementer : 0x41
CPU architecture: 7
CPU variant     : 0x0
CPU part        : 0xd08
CPU revision    : 3

processor       : 2
model name      : ARMv7 Processor rev 3 (v7l)
BogoMIPS        : 108.00
Features        : half thumb fastmult vfp edsp neon vfpv3 tls vfpv4 idiva idivt vfpd32 lpae evtstrm crc32 
CPU implementer : 0x41
CPU architecture: 7
CPU variant     : 0x0
CPU part        : 0xd08
CPU revision    : 3

processor       : 3
model name      : ARMv7 Processor rev 3 (v7l)
BogoMIPS        : 108.00
Features        : half thumb fastmult vfp edsp neon vfpv3 tls vfpv4 idiva idivt vfpd32 lpae evtstrm crc32 
CPU implementer : 0x41
CPU architecture: 7
CPU variant     : 0x0
CPU part        : 0xd08
CPU revision    : 3

Hardware        : BCM2711
Revision        : c03111
Serial          : 100000009771484f
Model           : Raspberry Pi 4 Model B Rev 1.1
```